wagtail/contrib/table_block/static/table_block/js/table.js

```diff
@@ -102,25 +92,34 @@ function initTable(id, tableOptions) {
     colHeaderCheckbox.change(function() {
         persist();
     });
-    
-    finalOptions.afterChange = cellEvent;
-    finalOptions.afterCreateCol = structureEvent;
-    finalOptions.afterCreateRow = structureEvent;
-    finalOptions.afterRemoveCol = structureEvent;
-    finalOptions.afterRemoveRow = structureEvent;
-    finalOptions.contextMenu = [
-        'row_above',
-        'row_below',
-        '---------',
-        'col_left',
-        'col_right',
-        '---------',
-        'remove_row',
-        'remove_col',
-        '---------',
-        'undo',
-        'redo'
-    ];
+
+    defaultOptions = {
+        afterChange: cellEvent,
+        afterCreateCol: structureEvent,
+        afterCreateRow: structureEvent,
+        afterRemoveCol: structureEvent,
+        afterRemoveRow: structureEvent,
+        contextMenu: [
+            'row_above',
+            'row_below',
+            '---------',
+            'col_left',
+            'col_right',
+            '---------',
+            'remove_row',
+            'remove_col',
+            '---------',
+            'undo',
+            'redo'
+        ]
+    };
+
+    if (dataForForm !== null && dataForForm.hasOwnProperty('data')) {
+        // Overrides default value from tableOptions (if given) with value from database
+        defaultOptions.data = dataForForm.data;
+    }
+
+    finalOptions = Object.assign({}, defaultOptions, tableOptions);
```

Comment:
The browser support for Object.assign isn't aligned (yet) with Wagtail's support target, so we can't use it without polyfills.

I don't think this JS is loading a polyfill at the moment – this will change once #3012 is merged.